doc/api/esm.md

```diff
@@ -938,6 +941,62 @@ export async function load(url, context, nextLoad) {
 In a more advanced scenario, this can also be used to transform an unsupported
 source to a supported one (see [Examples](#examples) below).
 
+#### `initialize()`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> The loaders API is being redesigned. This hook may disappear or its
+> signature may change. Do not rely on the API described below.
+
+* `data` {any} The data provided via `Module.register(loader, { data })`
+* Returns: {any} The data to be returned to the caller of `Module.register`
+
+The `initialize` hook provides a way to define a custom method of that runs
+in the loader's thread when the loader is initialized. This hook can send
+and receive data from a `Module.register` invocation, including ports and
+other transferrable objects.
+
```

Comment:
What happens if multiple loaders are registered and they each have `initialize` hooks? Presumably they aren’t chained; they just run in sequence, like `globalPreload`? We should probably document this, and include a test that multiple loaders’ `initialize` hooks behave as expected.